stw_potsdam/views.py

```diff
@@ -32,6 +34,7 @@
 
 cache = ct.TTLCache(maxsize=30, ttl=CACHE_TIMEOUT)
 
+swp_api = SWP_Webspeiseplan_API()
```

Comment:
If I understand correctly, we need to move the invocation of `SWP_Webspeiseplan_API()` into `canteen_menu_feed`, for example.
As a result, every scrape refreshes the list of menus.

I assume, if we stick with initialising variable `swp_api` only once during startup, we would run out of meals after 10 days. Would you agree?

Reply:
That is correct. It's not clear to me how openMensa interfaces the parser. I recommend to put swp_api into that function and set a timer for the actual renewal.

Reply:
> It's not clear to me how openMensa interfaces the parser

Does this help? https://docs.openmensa.org/parsers/understand/#how-openmensa-retrieves-canteen-menus-fetch

> set a timer for the actual renewal.

In the past it was OK to fetch the menu upon the first request. If the parser took to long, and openMensa detected a read timeout, openMensa would try again after 1h according to the documentation. The cache prevents that openMensa repeatedly incurs read timeouts: after the parser got one requests for the menu through, and understood the response of studentenwerk, we would have a cached response.